detection_rules/rule.py

```diff
@@ -515,6 +515,55 @@ def name(self) -> str:
     def type(self) -> str:
         return self.data.type
 
+    def _post_dict_transform(self, obj: dict) -> dict:
+        """Derive required fields from query AST."""
+
+        super()._post_dict_transform(obj)
+
+        self.add_related_integrations(obj)
+        self.add_required_fields(obj)
+        self.add_setup(obj)
+
+        return obj
+
+    def add_related_integrations(self, obj: dict) -> None:
+        """Add restricted field add_related_integrations to the obj"""
+        # field_name = "related_integrations"
+        ...
+
+    def add_required_fields(self, obj: dict) -> None:
+        """Add restricted field add_required_fields to the obj"""
+
+        field_name = "required_fields"
+        field_value = obj.get(field_name, [])
+        if self.check_restricted_field_version(field_name):
+
+            # validate the built field
+            self.validate_restricted_field_type(field_name, field_value)
+            obj.setdefault(field_name, field_value)
+
+    def add_setup(self, obj: dict) -> None:
+        """Add restricted field add_setup to the obj"""
+        # field_name = "setup"
+        ...
+
+    def check_restricted_field_version(self, field_name: str) -> bool:
```

Comment:
What's the use case for this?

If we have example fields:

```
field1 = (min=8.0)
field2 = (min=8.1, max=8.3)
field3 = (min=8.3)
```

and a global constant `GLOBAL_MIN_MAX = (min=8.0, max=8.4)` then all fields will be included.

with a global constant `GLOBAL_MIN_MAX = (min=8.1, max=8.4)` then field2 and fields3 are included.

Would we ever want to drop support for anything at the min or max? or Would we ever want to only build a field for certain versions even if they have a broader supported scope?

For example `GLOBAL_MIN_MAX_FIELD2 = (min=8.2)` would only build field2 on 8.2 stacks even though it is supported in kibana for versions 8.1 through 8.3, but what's the use case for ever doing that?

If there is a use case, then it may be better to make a GLOBAL lookup table which has optional global min max so that we don't have to add new global fields for each new field (assuming we don't want to be so restrictive).